### ISSUE TITLE
Password Reset form not working?

### ISSUE THREAD

**Issue by kernel-io** [team-alembic/ash_authentication_phoenix#541]
I have a weird issue where the submit event on the password reset form doesn't seem to be working.

```elixir
[error] GenServer #PID<0.1027.0> terminating
** (FunctionClauseError) no function clause matching in anonymous fn/1 in AshAuthentication.Phoenix.Components.Password.ResetForm.handle_event/3
    (ash_authentication_phoenix 2.2.1) deps/ash_authentication_phoenix/lib/ash_authentication_phoenix/components/password/reset_form.ex:149: anonymous fn(#Ash.Query<resource: Meow.Accounts.User, arguments: %{email: #Ash.CiString<"meowmeow@meow.com">}, filter: #Ash.Filter<email == #Ash.CiString<"meowmeow@meow.com">>>) in AshAuthentication.Phoenix.Components.Password.ResetForm.handle_event/3
    (ash_phoenix 2.1.8) lib/ash_phoenix/form/form.ex:1955: AshPhoenix.Form.submit/2
    (ash_authentication_phoenix 2.2.1) deps/ash_authentication_phoenix/lib/ash_authentication_phoenix/components/password/reset_form.ex:147: AshAuthentication.Phoenix.Components.Password.ResetForm.handle_event/3
    (phoenix_live_view 1.0.0-rc.7) lib/phoenix_live_view/channel.ex:741: anonymous fn/4 in Phoenix.LiveView.Channel.inner_component_handle_event/4
    (telemetry 1.3.0) /Volumes/Vault/SourceTrees/meow/platform/deps/telemetry/src/telemetry.erl:324: :telemetry.span/3
    (phoenix_live_view 1.0.0-rc.7) lib/phoenix_live_view/diff.ex:209: Phoenix.LiveView.Diff.write_component/4
    (phoenix_live_view 1.0.0-rc.7) lib/phoenix_live_view/channel.ex:662: Phoenix.LiveView.Channel.component_handle/4
    (stdlib 6.1.2) gen_server.erl:2345: :gen_server.try_handle_info/3
    (stdlib 6.1.2) gen_server.erl:2433: :gen_server.handle_msg/6
    (stdlib 6.1.2) proc_lib.erl:329: :proc_lib.init_p_do_apply/3
```

This is with a stock standard password strategy on the User resource.

```elixir
password :password do
  identity_field :email
  hash_provider AshAuthentication.ArgonProvider
  sign_in_tokens_enabled? true
  registration_enabled? false

  resettable do
    sender __MODULE__.Senders.PasswordResetEmail
    token_lifetime {1, :hours}
  end
end
```

**Comment by alexslade** [comment]
I'm also getting this, after updating to 2.2.0. 

@kernel-io If you go back to 2.1.11, that works for me as a temporary workaround. 

(I haven't yet investigating what's causing this though, sorry)

**Comment by zachdaniel** [comment]
This is fixed in `main` currently, will release it.

**Comment by zachdaniel** [comment]
Fixed in `2.3.0`
